proposals/2023/05-26_vela_ui_rewrite.md

```diff
@@ -0,0 +1,217 @@
+
+Elm is a fairly obscure language with a small community, anecdotally in our contributor's experience, anecdotally by way of these recent Hacker News threads about Elm [(1)](https://news.ycombinator.com/item?id=34746161) [(2)](https://news.ycombinator.com/item?id=35495910) [(3)](https://news.ycombinator.com/item?id=36272692) and statistically according to the [2022 State of JS Survey](https://2022.stateofjs.com/en-US/other-tools/#javascript_flavors) and [2023 Stackoverflow Survey](https://survey.stackoverflow.co/2023/#technology).
+
+Therefore we propose that a using more popular and widely used language such as TypeScript and a more popular and widely used framework such as React would significantly lower the barrier to entry to casual contributors to the Vela UI.
+
+<!-- We understand that Elm was originally for this codebase because... _please contribute this information_. -->
+
+**Please briefly answer the following questions:**
+
+1. Why is this required?
+
+<!-- Answer here -->
+
+Not required, however it should increase the number of contributors and contributions to the UI.
+
+2. If this is a redesign or refactor, what issues exist in the current implementation?
+
+<!-- Answer here -->
+
+- It is difficult to contribute to the UI, it is unlikely people who are interested in contributing know anything about Elm
+- In order to contribute it is likely the developer would need to learn a new language, which would put people off from contributing
+- The main project files are 5,000 - 7,000 lines of code each, which makes it difficult to find / figure out how things are tied together
+- ELM does not lend itself to code usability resulting in a copy / paste / rename / repurpose coding style.
+- There hasn't been any active development in the ELM framework for nearly 4 years, there are [many pull requests that have been sitting for 3+ years](https://github.com/elm/elm-lang.org/pulls?q=is%3Apr+is%3Aopen+sort%3Acreated-asc)
+
+3. Are there any other workarounds, and if so, what are the drawbacks?
+   <!-- Answer here -->
+
+   Alternatives to a redesign and rewrite could be...
+
+   1. maintaining the existing Elm codebase
+   2. creating and maintaining an extensive Elm-centric guide on maintenance activities, and new feature creation activities
+
+   In general, while possible, this may not necessarily increase the overall casual contributions.
+
+   However, with a redesign and rewrite...
+
+   With a focus on React and other widely popular and community accessible tooling, we think we can increase casual contribution.
+
+   This new rewrite will have:
+
+   - common React patterns
+   - common SPA patterns
+   - reusable components
+   - baseline guidance for creating new pages, widgets & components
+   - baseline guidance for introducing new dependencies
+
+4. Are there any related issues? Please provide them below if any exist.
```

Comment:
I'm not necessarily looking for Zac and Ryan to answer this question but it gets to the heart of all the areas of Vela... what criteria should we use to prioritizing rewriting any component? I like this question because it removes the immediacy of this proposal and allows us to take a better, unbiased look at the various factors.

I'd love for everyone to reply to this section...